bindings/python/src/session.cpp

```diff
@@ -206,68 +206,123 @@ namespace
 
     void dict_to_add_torrent_params(dict params, add_torrent_params& p)
     {
-        // torrent_info objects are always held by a shared_ptr in the python binding
-        if (params.has_key("ti") && params.get("ti") != boost::python::object())
+        list iterkeys = (list)params.keys();
+        int const len = int(boost::python::len(iterkeys));
+        for (int i = 0; i < len; i++)
```

Comment:
in python itself you can iterate over (key, value) - pairs, I would imagine it's possible in boost.python too.